iothub/service/src/Http/HttpMessageHelper.cs

```diff
@@ -92,7 +92,7 @@ internal static void ConditionallyInsertETag(HttpRequestMessage requestMessage,
             {
                 // "Perform this operation only if the entity is unchanged"
                 // Sends the If-Match header with a value of the ETag.
-                string escapedETag = EscapeETag(eTag.ToString());
+                string escapedETag = eTag.ToString("H");
```

Comment:
From Azure.Core:
"Valid values are "G" for standard format and "H" for header format."

https://github.com/Azure/azure-sdk-for-net/blob/9c6238e0f0dd403d6583b56ec7902c77c64a2e37/sdk/core/Azure.Core/src/ETag.cs#L87-L114